### ISSUE TITLE
@Html.DxaEntity() overrides original entity 

### ISSUE THREAD

**Issue by OskarKlintrot** [sdl/dxa-web-application-dotnet#23]
[`MvcHtmlString DxaEntity(this HtmlHelper htmlHelper, EntityModel entity, string viewName, int containerSize = 0)`](https://github.com/sdl/dxa-web-application-dotnet/blob/master/Sdl.Web.Mvc/Html/HtmlHelperExtensions.cs#L321-L336) overrides the original entity `viewName`. Is this done on purpose and if so, why? I tried to use the html helper function to use the same component twice due to how the CSS looks (the project is a migration where we want to reuse the HTML and CSS, it gets a bit weird).

Since it overrides the entity the code ends up something like this:

``` CSHARP
@Html.DxaEntity(Model.Entities.Single(m => m.MvcData.ViewName == "OriginalView"), "NewView")
@Html.DxaEntity(Model.Entities.Single(m => m.MvcData.ViewName == "NewView"), "EvenNewerView")
```

If it left the original entity alone the code could just be this:

``` CSHARP
@{
  var reuseMe = Model.Entities.Single(m => m.MvcData.ViewName == "OriginalView");
}
@Html.DxaEntity(reuseMe, "OneView")
@Html.DxaEntity(reuseMe, "AnotherView")
```

I get the feeling that I'm not really using it as intended...

**Comment by rpannekoek** [comment]
Current behavior is not intentional, no.  The idea is that you can render an Entity Model with another View than what the Entity Model's MVC data says (for whatever reason). This was intended to be a temporary override, though.

Can you further clarify why you want to render the same Entity Model with multiple different (hard-coded) Views, though?

**Comment by OskarKlintrot** [comment]
The HTML must look the same as the old site and there is two entities within the same div-tag. This to be able to reuse the CSS. The way I do it now is by simply using `@Html.Partial()`, so the code looks something like this now:

```HTML
@model RegionModel
@{ 
    var header = Model.Entities.Single(m => m.MvcData.ViewName == "Header");
}
<div class="header">
  <div class="menu">
    <div class="menu-content">
      @Html.DxaEntity(Model.Entities.Single(m => m.MvcData.ViewName == "Navigation"))
    </div>
    @Html.Partial("_Utility", header)
  </div>
  @Html.Partial("_Logo", header)
</div>
```

If there is a better way doing it I'm all ears!

**Comment by rpannekoek** [comment]
Interesting! It looks like you need nested Regions. :-)
However, since you're already finding the "Header" Entity by it's (original) View name first, would you actually suffer from the quirk/bug that it affects the MVC data of the Entity?  That is: will above Region View not work properly if you use `@Html.DxaEntity` with View override instead of `@Html.Partial` ?
Or are you also rendering the entity with its original View somewhere else?

**Comment by OskarKlintrot** [comment]
I tried that but no, since it is just referencing to the entity it wont help. So the following code wont work:

```html
@model RegionModel
@{
    var header = Model.Entities.Single(m => m.MvcData.ViewName == "Header");
}
<div class="header">
    <div class="menu">
        <div class="menu-content">
            @Html.DxaEntity(Model.Entities.Single(m => m.MvcData.ViewName == "Navigation"))
        </div>
        @Html.DxaEntity(header, "_Utility") @* This changes the MvcData.ViewName *@
    </div>
    @Html.DxaEntity(header, "_Logo") @* Argument Null Exception *@
</div>
```

This code will work though:

```html
@model RegionModel
<div class="header">
    <div class="menu">
        <div class="menu-content">
            @Html.DxaEntity(Model.Entities.Single(m => m.MvcData.ViewName == "Navigation"))
        </div>
        @Html.DxaEntity(Model.Entities.Single(m => m.MvcData.ViewName == "Header"), "_Utility")
    </div>
    @Html.DxaEntity(Model.Entities.Single(m => m.MvcData.ViewName == "_Utility"), "_Logo")
</div>
```

**Comment by rpannekoek** [comment]
TSI-1767 (internal issue ID for tracking purposes)

**Comment by rpannekoek** [comment]
Fixed in DXA 1.6.
